presto-main/src/main/java/com/facebook/presto/util/maps/IdentityLinkedHashMap.java

```diff
@@ -132,12 +168,156 @@ public int size()
                     return (Entry<K, V>) new AbstractMap.SimpleEntry<>(key, e.getValue());
                 }).iterator();
             }
-
-            @Override
-            public int size()
-            {
-                return delegate.size();
-            }
         };
     }
+
+    public abstract class IterateOnlySetView<E>
+            extends IterateOnlyCollectionView<E>
+            implements Set<E>
+    {
+        private IterateOnlySetView() {}
+    }
+
+    public abstract class IterateOnlyCollectionView<E>
+            implements Collection<E>
+    {
+        private IterateOnlyCollectionView() {}
+
+        @Override
+        public int size()
```

Comment:
Not sure if actually used, but `newSetFromMap(ILHM)` exposes this method, so cool if impl'd. Same for `isEmpty()`.